api/server/router/distribution/distribution_routes.go

```diff
@@ -122,7 +122,7 @@ func (s *distributionRouter) getDistributionInfo(ctx context.Context, w http.Res
 		var platform v1.Platform
 		if err == nil {
 			err := json.Unmarshal(configJSON, &platform)
-			if err == nil {
+			if err == nil && (platform.OS != "" || platform.Architecture != "") {
```

Comment:
The only time I actually saw one of Architecture or OS unset was for
Windows images, but that has since been fixed. I'm not sure if other less common platforms still have that issue. So far I haven't seen an image that would fail this check, so it should be fine.